core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java

```diff
@@ -148,7 +148,20 @@ public static boolean updateSegmentStatus(List<SegmentUpdateDetails> updateDetai
           mergeSegmentUpdate(isCompaction, oldList, newBlockEntry);
         }
 
-        segmentUpdateStatusManager.writeLoadDetailsIntoFile(oldList, updateStatusFileIdentifier);
+        List<SegmentUpdateDetails> updateDetailsValidSeg = new ArrayList<>();
```

Comment:
Map<String, LoadMetadataDetails>  details <- segmentUpdateStatusManager.getLoadMetadataDetails()

for (SegmentUpdateDetails updateDetail : oldList) {
     if(details.contains(updateDetail) {
     }
}

Reply:
@marchpure OK, use a hashset to check